packages/ember-routing/lib/system/route.js

```diff
@@ -1631,16 +1632,15 @@ let Route = EmberObject.extend(ActionHandler, Evented, {
 
     return {
       find(name, value) {
-        let modelClass = owner._lookupFactory(`model:${name}`);
-
+        let modelClass = owner.factoryFor(`model:${name}`);
         assert(
           `You used the dynamic segment ${name}_id in your route ${routeName}, but ${namespace}.${StringUtils.classify(name)} did not exist and you did not override your route's \`model\` hook.`, !!modelClass);
 
         if (!modelClass) { return; }
 
-        assert(`${StringUtils.classify(name)} has no method \`find\`.`, typeof modelClass.find === 'function');
+        assert(`${StringUtils.classify(name)} has no method \`find\`.`, typeof modelClass.class.find === 'function');
 
-        return modelClass.find(value);
+        return modelClass.class.find(value);
```

Comment:
In a future iteration, we should consider deprecating this feature. it is... dubious at best (and was my fault...)